ENIGMAsystem/SHELL/Universal_System/Resources/sprites.cpp

```diff
@@ -0,0 +1,258 @@
+}
+
+int sprite_add(std::string filename, int imgnumb, bool transparent, bool smooth, int x_offset, int y_offset, bool mipmap) {
+  return sprite_add(filename, imgnumb, false, transparent, smooth, false, x_offset, y_offset, mipmap);
+}
+
+bool sprite_replace(int ind, std::string filename, int imgnumb, bool precise, bool transparent, bool smooth, bool preload,
+                    int x_offset, int y_offset, bool free_texture, bool mipmap) {
+  if (free_texture) sprites.get(ind).FreeTextures();
+  return (sprites.replace(ind, sprite_add_helper(filename, imgnumb, precise, transparent, smooth, preload, x_offset, y_offset, mipmap)) != -1);
+}
+
+bool sprite_replace(int ind, std::string filename, int imgnumb, bool transparent, bool smooth, int x_offset, int y_offset,
+                    bool free_texture, bool mipmap) {
+  return sprite_replace(ind, filename, imgnumb, false, transparent, smooth, false, x_offset, y_offset, free_texture, mipmap);
+}
+
+bool sprite_exists(int spr) {
+  return sprites.exists(spr);
+}
+
+void sprite_delete(int ind, bool free_texture = true) {
+  if (free_texture) sprites.get(ind).FreeTextures();
+  sprites.destroy(ind);
+}
+
+int sprite_duplicate(int ind) {
+  return sprites.duplicate(ind);  
+}
+
+void sprite_assign(int ind, int copy_sprite, bool free_texture = true) {
+  if (free_texture) sprites.get(ind).FreeTextures();
+  Sprite copy = sprites.get(copy_sprite);
+  sprites.assign(ind, std::move(copy));
+}
+
+// NOTE: I doubt this method actually works properly on sprites of differing sizes
+// testing needed 
+void sprite_merge(int ind, int copy_sprite) {
+  Sprite& spr1 = sprites.get(ind);
+  const Sprite& spr2 = sprites.get(copy_sprite);
+  
+  // FIXME: this will break when we add functionality for removing subimages
+  for (size_t i = 0; i < spr2.SubimageCount(); ++i) {
+    spr1.AddSubimage(spr2.GetSubimage(i));
+  }
+   
+  spr1.width = std::max(spr1.width, spr2.width);
+  spr1.height = std::max(spr1.height, spr2.height);
+}
+
+void sprite_set_offset(int ind, int xoff, int yoff) {
+  Sprite& spr = sprites.get(ind);
+  spr.xoffset = xoff;
+  spr.yoffset = yoff;
+}
+
+// FIXME: free_texture unused
+void sprite_set_alpha_from_sprite(int ind, int copy_sprite, bool free_texture) {
+
+  Sprite& spr = sprites.get(ind);
+  const Sprite& spr_copy = sprites.get(copy_sprite);
+  
+  // FIXME: this will break when we add functionality for removing subimages
+  for (size_t i = 0; i < spr.SubimageCount(); i++)
+    enigma::graphics_replace_texture_alpha_from_texture(spr.GetTexture(i), spr_copy.GetTexture(i % spr_copy.SubimageCount()));
+}
+
+
+// FIXME: This sets the mode but the mode is currently meaningless
+void sprite_set_bbox_mode(int ind, int mode) {
+  sprites.get(ind).bbox_mode = mode;
+}
+
+void sprite_set_bbox(int ind, int left, int top, int right, int bottom) {
+  sprites.get(ind).bbox = {left, top, right - left, bottom - top };
```

Comment:
This is kind of error-prone.

Reply:
error prone how?

Reply:
If the field order changes in any way, it'll break with no compiler errors.

Reply:
idk what you expect me to do about that. It's gm's api not ours